cmd/edgex-secrets-config/README.md

```diff
@@ -0,0 +1,179 @@
+
+    * **--incert** _/path/to/certchain_ (required)
+
+      Path to TLS leaf certificate (PEM-encoded x.509) (the file extension is arbitrary).
+      If intermediate certificates are required to chain to a certificate authority,
+      these should also be included.
+      The root certificate authority should not be included.
+
+    * **--inkey** _/path/to/private\_key_ (required)
+
+      Path to TLS private key (PEM-encoded).
+
+  * **adduser**
+
+    Create an API gateway user using specified token type. Requires additional arguments:
+
+    * **--token-type** jwt | oauth2 (required)
+    
+      Create user using either the JWT or OAuth2 authentication plugin.
+      This value must match the configured authentication plugin
+      (`KongAuth.Name` in security-proxy-setup's `configuration.toml`).
+
+    * **--user** _username_ (required)
+
+      Username of the user to add.
+
+    * **--group** _group_ (optional)
+
+      Group to which the user belongs, defaults to &quot;admin&quot;.
+      This should be the group associated with the route ACL
+      (`KongAuth.WhiteList` in security-proxy-setup's `configuration.toml`).
+      (Note that edgex-secrets-config shares the same configuration as security-proxy-setup
+      as they both configure the EdgeX API gateway.)
+
+
+    The following options are used when token-type == "jwt":
+
+    * **--algorithm** RS256 | ES256 (required for JWT method)
+
+      Algorithm used for signing the JWT.
+      (See [RFC 7518](https://tools.ietf.org/html/rfc7518#section-3.1) for a list of signing algorithms.)
+
+    * **--public\_key** _/path/to/public\_key_ (required for JWT tokens)
+
+      Public key (in PEM format) used to validate the JWT.
+      (Not an x.509 certificate.)
+      This key is assumed to have been pre-created using some external mechanism such as a TPM, HSM, openssl, or other method.
+
+    * **--id** _key_ (optional)
+
+      Optional user-specified &quot;key&quot; used for linkage with an incoming JWT via Kong&#39;s config.key\_claim\_name setting (defaults to &quot;iss&quot; field).
+      See
+      [Kong documentation for JWT plugin](https://docs.konghq.com/hub/kong-inc/jwt/#craft-a-jwt-with-publicprivate-keys-rs256-or-es256)
+      for an example of how this parameter is used.
+
+
+    The following options are used when token-type == "oauth2":
+
+    * **--client\_id** (optional)
+
+      Optional manually-specified OAuth2 client_id.  Will be generated if not present.  Equivalent to a username.
+
+    * **--client\_secret** (optional)
+
+      Optional manually-specified OAuth2 client_secret.  Will be generated if not present.  Equivalent to a password.
+
+    * **--redirect\_uris** _url\_for\_browser\_redirection_ (required for oauth2 tokens)
+
+      OAuth2 redirect URL for browser-based users.  Not currently used by EdgeX.
+
+    Upon completion, the command outputs a JSON structure containing the client_id and client_secret.
```

Comment:
@tonyespy , @bnevis-i , we should not need to hard code setting the loglevel to ERROR if we are setting it to ERROR as the default in the configuration file. Not sure why setting it to ERROR in the config file isn't working for @tonyespy . I try one of the other services and see it works properly there.

Reply:
So with App Service configurable, I get the following when loglevel is set to ERROR in config file.

```
level=INFO ts=2020-10-30T18:46:46.600898742Z app=AppService-<profile> source=sdk.go:339 msg="Starting AppService-sample 0.0.0 "
level=INFO ts=2020-10-30T18:46:46.60216181Z app=AppService-sample source=config.go:193 msg="Loaded configuration from ./res/sample/configuration.toml"
level=ERROR ts=2020-10-30T18:46:46.610667301Z app=AppService-sample source=logger.go:74 msg="logTarget cannot be blank, using stdout only"
```

This is because bootstrap logs to first two message before is loads the configuration and creates the properly configured logging client.

Reply:
What do y'all want to do about it?  Should I put the output parameters back in since there doesn't seem to be a way to make the tool silent?

Reply:
This is resolved by latest go-mod-bootstrap.